test/Greentube.Monitoring.Tests/ResourceMonitorTests.cs

```diff
@@ -197,6 +197,37 @@ public void Verify_EventHandlerThrows_CriticalLogEntry()
             }
         }
 
+        [Theory]
+        [InlineData(true, LogLevel.Error)]
+        [InlineData(false, LogLevel.Warning)]
+        public void Verify_ResourceStateDown_LogEntryLevelBasedOnCritical(bool isCritical, LogLevel expectedLogLevel)
+        {
+            var expectedException = new ArithmeticException();
+            _fixture.IsCritical = isCritical;
+            _fixture.HealthCheckStrategy.Check(Arg.Any<CancellationToken>()).Throws(expectedException);
+            using (var sut = _fixture.GetSut())
+            {
+                sut.Verify();
+                _fixture.Logger
+                    .Received(1)
+                    .Log(expectedLogLevel, Arg.Any<EventId>(), Arg.Any<object>(), Arg.Any<ArithmeticException>(), Arg.Any<Func<object, Exception, string>>());
+            }
+        }
+
+        [Fact]
+        public void Verify_ResourceStateUp_LogEntryLevelExpected()

```

Comment:
I would say it is not LogEntryLevelExpected.
We don't expect a presence of Level but we expect the fact that it was logged with Trace.
So should be _LoggedAsTrace or even combined with previous test using third parameter

Reply:
fixed